sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -366,7 +366,7 @@ public static string CanonicalizeResourcePath(Uri endpoint, string resourcePath,
                 }
             }
 
-            var canonicalizedResourcePath = AWSSDKUtils.JoinResourcePathSegments(encodedSegments, false);
+            var canonicalizedResourcePath = AWSSDKUtils.JoinResourcePathSegments(encodedSegments, true);
```

Comment:
Thanks for opening up a PR. The issue in question #1943 is complaining that we are encoding special characters twice. In the example listed (`"/%3A"`), `CanonicalizeResourcePath` returns `"/%25253A"` while the user is saying it should return `"%253A`. Changing this value from false to true doesn't fix the issue, it just completely ignores uri encoding altogether.

Whether or not this is a bug or by design requires more investigation but this PR needs tests and the solution does not fix the underlying issue so I'm going to go ahead and close the PR. I will leave the issue open as we investigate this further.